### PR TITLE
Rewrite: breakOut (fix #46)

### LINKED ISSUE #46: Rewrite breakout
seen in:
https://github.com/kxbmap/configs/pull/30/files#diff-8416f81bdac5bf53701680a5968a0f9eL226

input:

```scala
import scala.collection.breakOut
val map : Map[Int,String] = List("London", "France").map(x => (x.length, x))(breakOut)
```
output:

```scala
val map : Map[Int,String] = List("London", "France").map(x => (x.length, x)).toMap
```

### PR DESCRIPTION
**Pull request by MasseGuillaume**

### CODE REVIEW

File: scalafix/input/src/main/scala/fix/BreakoutSrc.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+package fix
+
+import scala.collection.breakOut
+
+object BreakoutSrc {
+  val xs = List(1, 2, 3)
+
+  xs.collect{ case x => x }(breakOut): Set[Int]
+  xs.flatMap(x => List(x))(breakOut): Set[Int]
```
Comment (julienrf) [inline scalafix/input/src/main/scala/fix/BreakoutSrc.scala]: Can you also try with a sorted result? `SortedSet[Int]` for instance.

File: scalafix/input/src/main/scala/fix/BreakoutSrc.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  xs.map(_ + 1)(breakOut): Set[Int]
+  xs.reverseMap(_ + 1)(breakOut): Set[Int]
+  xs.scanLeft(0)((a, b) => a + b)(breakOut): Set[Int]
+  xs.union(xs)(breakOut): Set[Int]
+  xs.updated(0, 1)(breakOut): Set[Int]
+  xs.zip(xs)(breakOut): Array[(Int, Int)]
```
Comment (julienrf) [inline scalafix/input/src/main/scala/fix/BreakoutSrc.scala]: Can you try with a `Map[Int, Int]`?

File: scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala [hunk 3]
```diff
@@ -500,6 +554,58 @@ case class Scalacollectioncompat_newcollections(index: SemanticdbIndex)
...
+      case Term.Apply(ap @ Term.ApplyInfix(lhs, operators(op), _, List(rhs)), List(breakOut(bo))) =>
+        val subject =
+          if(isLeftAssociative(op)) lhs
+          else rhs
+
+        val intermediate = 
+          op match {
+            case operatorsIterator(_) => "iterator"
+            case operatorsView(_)     => "view"
+            case _                    => throw new Exception("impossible")
```
Comment (julienrf) [inline scalafix/rules/src/main/scala/fix/Scalacollectioncompat_newcollections.scala]: Maybe add a comment referring to the definition of `operatorSymbols`, explaining why that makes that path impossible to reach?

**Comment by MasseGuillaume** [comment]
@julienrf all comments addressed.

**Pull request merged**